updateStableIds/src/main/java/org/reactome/release/update_stable_ids/UpdateStableIds.java

```diff
@@ -0,0 +1,170 @@
+		sliceInstances.addAll(eventInstances);
+		sliceInstances.addAll(physicalEntityInstances);
+		
+		int incrementCount = 0;
+		logger.info("Total instances to check: " + sliceInstances.size());
+		for (GKInstance sliceInstance : sliceInstances) {
+			logger.info("Checking " + sliceInstance);
+			GKInstance gkCentralInstance = (GKInstance) dbaGkCentral.fetchInstance(sliceInstance.getDBID());
+			GKInstance prevSliceInstance = (GKInstance) dbaPrevSlice.fetchInstance(sliceInstance.getDBID());
+			// Check if instance is new and that it exists on gkCentral (they could be deleted)
+			if (prevSliceInstance != null && gkCentralInstance != null) {
+				
+				// Compare number of 'Modified' instances between slices
+				Collection<GKInstance> sliceInstanceModified = sliceInstance.getAttributeValuesList(modified);
+				Collection<GKInstance> prevSliceInstanceModified = prevSliceInstance.getAttributeValuesList(modified);
+				if (sliceInstanceModified.size() > prevSliceInstanceModified.size()) {
+					// Make sure StableIdentifier instance exists
+					if (sliceInstance.getAttributeValue(stableIdentifier) != null && gkCentralInstance.getAttributeValue(stableIdentifier) != null) {
+						logger.info("\tIncrementing " + sliceInstance.getAttributeValue(stableIdentifier));
+						incrementStableIdentifier(sliceInstance, dbaSlice, sliceIE);
+						incrementStableIdentifier(gkCentralInstance, dbaGkCentral, gkCentralIE);
+						incrementCount++;
+					}
+				} else {
+					if (sliceInstanceModified.size() == prevSliceInstanceModified.size()) {
+//						System.out.println("No change between releases");
+					} else {
+						logger.warn("Unexpected: Number of 'Modified' instances in [" + sliceInstance.getDBID() + "] is fewer than in previous release");
+					}
+				}
+				
+				// Instances that have been updated already during the current release will have their 'releaseStatus' attribute equal to 'UPDATED'.
+				// This will make sure that StableIDs are only updated once per release.
+				try {
+					if (isUpdated(sliceInstance, prevSliceInstance, dbaPrevSlice)) {
+						String releaseStatusString = (String) sliceInstance.getAttributeValue(releaseStatus);
+						String updated = "UPDATED";
+						if (releaseStatus != updated) {
+							sliceInstance.addAttributeValue(releaseStatus, updated);
+							sliceInstance.addAttributeValue(modified, sliceIE);
+							dbaSlice.updateInstanceAttribute(sliceInstance, releaseStatus);
+							dbaSlice.updateInstanceAttribute(sliceInstance, modified);
+						} else {
+							logger.info("StableIdentifer has already been updated during this release");
+						}
+					}
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		}
+
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.commit();
+		}
+		logger.info("Commiting all changes in " + dbaGkCentral.getDBName());
+		dbaGkCentral.commit();
+		logger.info(incrementCount + " Stable Identifiers were updated");
```

Comment:
Would it also be useful to gather on the number of Stable Identifiers that were checked but did not need updating? Not sure if users would find that informative. Might not be necessary.